pkg/build/pipelines/go/build.yaml

```
@@ -101,8 +101,22 @@ inputs:
       If true, "go mod tidy" will run before the build
     default: "false"
 
+  ignore-untracked-files:
+    description: |
+      If true, we will provide a gitignore that ignore all untracked files.
+    default: "true"
+
 pipeline:
   - runs: |
+
+      # We sometimes have stray files in our build environment
+      # That we don't neccesarily want surfaced in the build version
+      if [ "${{inputs.ignore-untracked-files}}" == "true" ]; then
+        mkdir -p ~/.config/git/
+        echo "**" > ~/.config/git/ignore
+        echo ".*" >> ~/.config/git/ignore
```

```suggestion
        echo '**' > ~/.config/git/ignore
        echo '.*' >> ~/.config/git/ignore
```

Might want to use single quote here to avoid any unwanted shell expansion.

Also, from the [documentation](https://git-scm.com/docs/gitignore#_pattern_format), I believe `**` already matches everything, so maybe we don't need `.*`?

---

Sounds good i'll check this.